Automattic/wp-calypso
# Site Profiler: Implement copy strategy for Header and Basic metrics

Resolves https://github.com/Automattic/dotcom-forge/issues/7435
## Proposed Changes

* Create four different categories to display the results header
* Create a copies component that returns the translated copies, titles, and CTA URLs for the basic cards
* Generalizes the Basic Card component to use the list of copy-objects returned by the copies component above
* Add all the copies for the main title and basic metric cards as currently defined in the P2 p9Jlb4-bWg-p2

> [!WARNING]  
> The copies' content is still under discussion, and many of them will change. They are an initial prototype, and they have been added before review and approval to progress with the development iteratively.

> [!WARNING]  
> The metrics that are considered as `needs-improvement` by the google docs are considered to be not good in this implementation. This can change after further discussions and reviews.


## Why are these changes being made?


* To add the copies required for the Site Profiler tool

## Testing Instructions



* Apply this patch
* Select four different sites with the following characteristics. I have listed some examples here 34a1c-pb/#plain
  * Performs poorly, and it is not hosted on WP.com
  * Performs well, and it is not hosted on WP.com
  * Performs well, and it is hosted on WP.com
  * Performs poorly, and it is hosted on WP.com
* For each of the sites above
* Check that the title copy and header background correspond with the expected one as described in the P2 p9Jlb4-bWg-p2
* Check that all the values in each of the Basic Metric cards are the expected ones as described in the P2 p9Jlb4-bWg-p2
* Additionally, you can manually tweak [this condition](https://github.com/Automattic/wp-calypso/blob/1c6338144db69777cb21dd42e2a74e08e8ed0947/client/site-profiler/components/basic-metrics/index.tsx#L42) to make all the results bad or all of them good for the 4 different categories, so all the texts can be checked without needing to find a site that fits into that category for that metric.


| Category | Screenshot |
|--------|--------|
| Non-WPCOM high performer | ![nwpcom-good](https://github.com/Automattic/wp-calypso/assets/11555574/3ad557ee-da48-4017-8ade-22c3aec1b05e) |
| Non-WPCOM low performer | ![nwpcom-bad](https://github.com/Automattic/wp-calypso/assets/11555574/bb580884-a65d-4b01-bb2a-2069b04d4660) |
| WPCOM high performer | ![wpcom-good](https://github.com/Automattic/wp-calypso/assets/11555574/e4478ec8-effe-4fe3-aeaf-c133fcf40997) |
| WPCOM low performer | ![wpcom-bad](https://github.com/Automattic/wp-calypso/assets/11555574/4e26ea55-217d-4d4f-a7bf-d6163f5696e0) | 

## Pre-merge Checklist



~~- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?~~
~~- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?~~
